### PR TITLE
Cleanup product credit

### PR DESCRIPTION
**Pull request by nickpell**
Addresses the backlog item from http://manage.dimagi.com/default.asp?221644#BugEvent.1112132.  Also prevents a CommTrack client who purchases plan credit from losing access to the plan credit if they switch to a CommCare subscription.

:white_check_mark: from Ops

:fish: or :european_post_office: 

@benrudolph cc: @czue

### CODE REVIEW

File: corehq/apps/accounting/migrations/0023__simplify__credit_line__product_type.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    dependencies = [
+        ('accounting', '0022_bootstrap_prbac_roles'),
+    ]
+
+    operations = [
+        HqRunPython(confirm_no_mismatched_subscription_credit),
+        HqRunPython(confirm_no_mismatched_account_credit_can_be_invoiced_automatically),
+        HqRunPython(set_credit_line_product_types_to_any),
+        migrations.AlterField(
```
Comment (czue) [inline corehq/apps/accounting/migrations/0023__simplify__credit_line__product_type.py]: :+1: this is a great pattern

**Comment by czue** [comment]
everything looks good to me. think I should let someone with billing context double check everything though.

**Comment by benrudolph** [comment]
seems right to me!

File: corehq/apps/accounting/migrations/0023__simplify__credit_line__product_type.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+For production environments, reconcile credit lines on a case by case basis.
+
+For dev environments, simply delete problematic credit lines:
+
+from datetime import date
+from django.db.models import Q
+from corehq.apps.accounting.models import CreditAdjustment, CreditLine, Subscription
+previous_invoicing_date = date(date.today().year, date.today().month, 1)
+subscriptions_in_future_invoicing_periods = Subscription.objects.exclude(
```
Comment (gcapalbo) [inline corehq/apps/accounting/migrations/0023__simplify__credit_line__product_type.py]: @nickpell i noticed Subscription overrides the default objects model manager. should we still keep the default model manager under a new name, like all_objects or something similar? i kept running this on staging and it wasn't deleting anything until i realized that was the problem

Comment (nickpell) [inline corehq/apps/accounting/migrations/0023__simplify__credit_line__product_type.py]: oops, should have ignored suppressed subscriptions

https://github.com/dimagi/commcare-hq/pull/11272

**Pull request merged**